drink/src/runtime/pallet_contracts_debugging/intercepting.rs

```diff
@@ -0,0 +1,25 @@
+use pallet_contracts::debug::{CallInterceptor, ExecResult, ExportedFunction};
+use parity_scale_codec::{Decode, Encode};
+
+use crate::runtime::{
+    pallet_contracts_debugging::{runtime::contract_call_debugger, DrinkDebug},
+    AccountIdFor, Runtime,
+};
+
+impl<R: Runtime> CallInterceptor<R> for DrinkDebug {
+    fn intercept_call(
+        contract_address: &AccountIdFor<R>,
+        entry_point: &ExportedFunction,
+        input_data: &[u8],
+    ) -> Option<ExecResult> {
+        // Pass the data to the runtime interface. The data must be encoded (only simple types are
+        // supported).
+        let intercepting_result = contract_call_debugger::intercept_call(
+            contract_address.encode(),
+            matches!(*entry_point, ExportedFunction::Call),
+            input_data.to_vec(),
+        );
+
+        Decode::decode(&mut intercepting_result.as_slice()).expect("Decoding should succeed")
+    }
+}
```

Comment:
If the `ContractcallDebugger` was part of Substrate, alongside the `CallInterceptor`, there could be a generic implementation 
```rust
impl<T, R: Runtime> CallInterceptor<R> for T { ... }
```
since, unless I misunderstand sth (which is highly probable), there's nothing about `DrinkDebug` that makes it impossible to implement generically.

Reply:
yeah, that's right; but at least for a few weeks it has to reside here, until they agree to accept this code in pallet and they review my PR xD

Reply:
Sure, understand it. It was also a question to clear up _my_ understanding about the system and where each piece (could) lie.